prog/postgres/postgres_server_nexus.rb

```diff
@@ -16,14 +16,18 @@ def self.assemble(resource_id:, timeline_id:, timeline_access:, representative_a
       ubid = PostgresServer.generate_ubid
 
       postgres_resource = PostgresResource[resource_id]
+      # For read replicas, we upgrade by replacing servers instead of in-place
+      # upgrade.
+      server_version = postgres_resource.read_replica? ? postgres_resource.target_version : postgres_resource.version
```

Comment:
At this point, is the parent already upgraded? Should we not wait for the parent to be completely upgraded before calling this? If we do that already, isn't the `postgres_resource.version` and `target_version` are the same?